modules/core/src/main/java/org/apache/ignite/internal/util/VersatileReadWriteLock.java

```diff
@@ -390,15 +405,15 @@ private void notifyReadLockSolicitors() {
                 break;
             }
 
+            iterator.remove();
```

Comment:
Same here

Reply:
Futures represent acqusition intents. We have just acquired the write lock for this intent, so we should remove it from the waiting set (as its acqusition request has been satisfied)

Reply:
I implemented this change as well